src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer_Core.cs

```diff
@@ -43,7 +40,9 @@ protected AbstractBuiltInCodeStyleDiagnosticAnalyzer(ImmutableArray<DiagnosticDe
         SupportedDiagnostics = supportedDiagnostics;
 
         Descriptor = SupportedDiagnostics[0];
-        Debug.Assert(!supportedDiagnostics.Any(descriptor => descriptor.CustomTags.Any(t => t == WellKnownDiagnosticTags.Unnecessary)) || this is AbstractBuiltInUnnecessaryCodeStyleDiagnosticAnalyzer);
+        Debug.Assert(
+            !supportedDiagnostics.Any(static descriptor => descriptor.CustomTags.Any(t => t == WellKnownDiagnosticTags.Unnecessary))
+            || this is AbstractBuiltInUnnecessaryCodeStyleDiagnosticAnalyzer or AbstractBuiltInCodeStyleDiagnosticAnalyzer);
```

Comment:
>AbstractBuiltInCodeStyleDiagnosticAnalyzer [](http://example.com/codeflow?start=80&length=42)

Noticed this because as soon as I changed to ImmutableArrays, I hit this assert. Previously, the ImmutableArray would end up having some entries overwrite eachother due to the DiagnosticDescriptor's Equals not using CustomTags in it's comparison.